File: src/types/notifikasi.ts
import type { RoleName } from "./auth";

export type TipeNotifikasi = "stok_menipis" | "obat_expired" | "sistem";

export interface Notifikasi {
  id: string;
  cabangId?: string;
  penggunaId?: string;
  tipe: TipeNotifikasi;
  judul: string;
  pesan?: string;
  referensiTabel?: string;
  referensiId?: string;
  isRead: boolean;
  targetRole?: RoleName;
  createdAt: string;
}

File: backend/app/models/Notifikasi.go
package models

import "time"

type Notifikasi struct {
	ID             string    `json:"id"`
	CabangID       *string   `json:"cabang_id"`
	PenggunaID     *string   `json:"pengguna_id"`
	TargetRole     *string   `json:"target_role"`
	Tipe           string    `json:"tipe"`
	Judul          string    `json:"judul"`
	Pesan          *string   `json:"pesan"`
	ReferensiTabel *string   `json:"referensi_tabel"`
	ReferensiID    *string   `json:"referensi_id"`
	IsRead         bool      `json:"is_read"`
	CreatedAt      time.Time `json:"created_at"`
}
